### DISCUSSION TITLE
Módulo 4 duda "For" y "In"

### DISCUSSION THREAD

**Discussion by CarlosHernand3z** [LaunchX-InnovaccionVirtual/CursoIntroPython#213] [Q&A]
Hola buenas tardes, no sé si sea que ando atareado con tantas actividades pero no encuentro información que nos enseñen sobre "For" and "In" en forma de texto o por lo menos no en la "Guía del módulo 4" que es donde nos dan un breve resumen de lo que vamos aplicar en los ejercicios, solo encontré  información en el segundo live. ¿Hay algo de lo que me perdí? Me siento un poco fustrado porque no me gusta ver los ejercicios pero cuando miré para ver dije ah caray el "for" y el "in" no me lo habían explicado.  

¡Gracias de antemano!

PD: Creo que yo mismo me confundí solo jajaja ya resolví mi duda, me hace falta comprensión lectora, igual muchas gracias por sus respuestas todas me fueron de utilidad :)

**Comment by MarcosSilvaG** [comment]
te sugiero revisar en la [documentación oficial](https://docs.python.org/3.10/reference/compound_stmts.html#the-for-statement) del lenguaje :D o también en[ otras paginas web](https://j2logo.com/bucle-for-en-python/) para ampliar tu información acerca de un tema.
Animos!

**Reply by MarcosSilvaG** [reply]
básicamente el "in" es parte de la estructura del "for" que es una estructura de iteración (repeticiones) que en su declaración tiene la forma `for "target_list" in "expression_list":`

**Comment by doguedogue** [comment]
Puedes usarlo así:

`lista =[1,2,3,4]
for valor_obtenido in lista
    print (valor_obtenido)`

https://wiki.python.org/moin/ForLoop

**Comment by carlos-gmx3** [comment]
No lo hicieron pero bien puedes buscar la acerca del uso de for en internet
a mi me descontroló ya que estaba acostumbrado a declaralo en lenguajes como JS, arduino o C++
checa esta pagina https://j2logo.com/bucle-for-en-python/
Te comparto una parte del ejercicio y como yo lo interpreté

Primero dividí la cadena en un arreglo o array de varias cadenas partiendo la cadena original en cada punto que encuentre la funcion split
`sentences = text.split('.')`
sentences ya esta declarado ok? ahora, como parte del for se daclara una variable
cual será su valor? pues simplemente cuando for recorra el array en esta variable "sentence"
estará el contenido de cada elemento de la cadena, lo procesa con el codigo dentro del bucle
y cambia su valor por el suiguiente elemento del array... así hasta terminar con todos
`for sentence in sentences:
  //Codigo a ejecutar dentro del for es todo lo que tiene el siguiente nivel de indentación`

es mas, prueba el siguiente código y tranta de entenderlo por ti mismo
`text = """Interesting facts about the Moon. The Moon is Earth's only satellite. There are several interesting facts about the Moon and how it affects life here on Earth. 
On average, the Moon moves 4cm away from the Earth every year. This yearly drift is not significant enough to cause immediate effects on Earth. The highest daylight temperature of the Moon is 127 C."""`
`sentences = text.split('.')`
`print(text)`
`print(sentences)`
`for sentence in sentences:`
  `  print(sentence)` // no olvides agregar una tabulación o doble espacio a esta instrucción para que la ejecute el for, por alguna razón github no la está respetando
